students/14th/team_05/suhee-go/project_westagram/user/views.py

```diff
@@ -0,0 +1,89 @@
+import json
+import re
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
+from .models          import User
+
+class SignUpView(View):
+    def post(self,request):
+        data                = json.loads(request.body)
+
+        signup_name         = data["name"]
+        signup_email        = data["email"]
+        signup_phone        = data["phone_number"]
+        signup_password     = data["password"]
+
+        valid_name          = re.compile(r'[a-z0-9_.]{5,30}')
+        valid_email         = re.compile(r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+        vaild_phone         = re.compile(r'[:digit:]')
+        valid_password      = re.compile(r'[a-z0-9@#$]{8,12}')
+
+        validation_name     = re.search(valid_name, signup_name)
+        validation_email    = re.search(valid_email, signup_email)
+        validation_phone    = re.search(vaild_phone, signup_phone)
+        validation_password = re.search(valid_password, signup_password)
+
+        try:
+            # vaildation check
+            if validation_name :
+                pass
+            else:
+                return JsonResponse({"messgage" : "INVAILD_NAME"}, status = 400)
```

Comment:
```suggestion
            if not validation_name:
                return JsonResponse({"messgage" : "INVAILD_NAME"}, status = 400)
```
요렇게 진행하시면 훨씬 간결한 `if` 로직으로 같은 기능을 구현하실 수 있습니다 :)

Reply:
확인했습니다. 수정완료했습니다.